### PR TITLE
Fix autoloading libraries before they are needed

### PR DESCRIPTION
**Pull request by kke**
There were several leaks in the autoload system.

### Before

(with pry disabled from gemspec and `require` monkeypatched to print out requires)

```
$ bundle exec bin/pharos --version | wc -l
     908
$ PHAROS_NON_OSS=true bundle exec bin/pharos --version | wc -l
     916
```

### After

```
$ bundle exec bin/pharos --version | wc -l
      38
$ PHAROS_NON_OSS=true bundle exec bin/pharos --version | wc -l
      46
```

### CODE REVIEW

File: lib/pharos/transport/ssh.rb [hunk 1]
```diff
@@ -12,30 +12,35 @@ class SSH < Base
         Net::SSH::Authentication::KeyManagerError
       ].freeze
 
-      # @param host [String]
-      # @param opts [Hash]
-      def initialize(host, **opts)
-        super(host, opts)
-        @user = @opts.delete(:user)
+      def to_s
```
Comment (jakolehm) [inline lib/pharos/transport/ssh.rb]: Changes in this file are not relevant to this PR, right?

File: lib/pharos/core-ext/colorize.rb [hunk 2]
```diff
@@ -18,7 +25,7 @@ def self.enabled?
       end
 
       refine String do
-        Pastel::ANSI::ATTRIBUTES.each_key do |meth|
+        ATTRIBUTES.each do |meth|
```
Comment (jakolehm) [inline lib/pharos/core-ext/colorize.rb]: Is this really needed?

File: lib/pharos/transport/base.rb [hunk 3]
```diff
@@ -20,12 +20,10 @@ class Base
 
       attr_reader :host
 
-      # @param host [String]
-      # @param opts [Hash]
-      def initialize(host, **opts)
+      # @param host [Pharos::Configuration::Host]
+      def initialize(host)
```
Comment (jakolehm) [inline lib/pharos/transport/base.rb]: Not related to autoload.

File: lib/pharos/core-ext/colorize.rb (hunk 2 above)
Comment (kke) [reply lib/pharos/core-ext/colorize.rb]: Yes but the line below isn't anymore as `underscore` can be removed from the ATTRIBUTES array.

Alternatives would be to define each method delegator manually, add a `method_missing` delegator or going back to regular pastel.

File: lib/pharos/transport/base.rb (hunk 3 above)
Comment (kke) [reply lib/pharos/transport/base.rb]: We can bring back transport.rb and remove the requires.

File: lib/pharos/core-ext/colorize.rb (hunk 2 above)
Comment (jakolehm) [reply lib/pharos/core-ext/colorize.rb]: So what do we gain by copypasting Pastel attributes here?

Comment (kke) [reply lib/pharos/core-ext/colorize.rb]: We get them without loading pastel which loads `tty-color` and `equatable`.

**Pull request closed without merging**